File: Exemplos/App_IntentDigitalHUB_ReactNative/Elgin_IntentDigitalHUB_ReactNative-font/src/intentDigitalHubService/BALANCA/AbrirSerial.ts
import {BalancaCommand} from './BalancaCommand';

export class AbrirSerial extends BalancaCommand {
  readonly baudrate: number;
  readonly lenght: number;
  readonly parity: string;
  readonly stopbits: number;

  constructor(
    baudrate: number,
    lenght: number,
    parity: string,
    stopbits: number,
  ) {
    super('AbrirSerial');
    this.baudrate = baudrate;
    this.lenght = lenght;
    this.parity = parity;
    this.stopbits = stopbits;
  }

  functionParameters(): object {
    return {
      baudrate: this.baudrate,
      lenght: this.lenght,
      parity: this.parity,
      stopbits: this.stopbits,
    };
  }
}

File: Exemplos M10/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/IntentServices/Balanca/Commands/AbrirSerial.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Xamarin_Forms_Intent_Digital_Hub.IntentServices.Balanca.Commands
{
    class AbrirSerial : BalanceCommand
    {
        readonly int baudrate;
        readonly int lenght;
        readonly char parity;
        readonly int stopbits;

        public AbrirSerial(int baudrate, int lenght, char parity, int stopbits) : base("AbrirSerial")
        {
            this.baudrate = baudrate;
            this.lenght = lenght;
            this.parity = parity;
            this.stopbits = stopbits;
        }

        protected override string FunctionParameters()
        {
            return "\"baudrate\"" + ":" + this.baudrate + "," +
                "\"lenght\"" + ":" + this.lenght + "," +
                "\"parity\"" + ":" + "\"" + this.parity + "\"" + "," +
                "\"stopbits\"" + ":" + this.stopbits;
        }

    }
}
